src/core/primitives/Menu/fragments/MenuPrimitivePortal.tsx

```
@@ -0,0 +1,27 @@
+'use client';
+import React, { useContext, useEffect, useState } from 'react';
+import Floater from '~/core/primitives/Floater';
+import MenuPrimitiveRootContext from '../contexts/MenuPrimitiveRootContext';
+
+function MenuPrimitivePortal({ children }: { children: React.ReactNode }) {
+    const context = useContext(MenuPrimitiveRootContext);
+    if (!context) return null;
+    const { isOpen } = context;
+    const [rootElementFound, setRootElementFound] = useState(false);
```

Biome (2.1.2)

[error] This hook is being called conditionally, but all hooks must be called in the exact same order in every component render.

Hooks should not be called after an early return.

For React to preserve state between calls, hooks needs to be called unconditionally and always in the same order.
See https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level

(lint/correctness/useHookAtTopLevel)